docs/workflows/zip-code-integration-template.md

```diff
@@ -0,0 +1,116 @@
+# Integration templates
+
+Developers can [extend](index.md) the plugin to meet uses cases that require custom logic. Often, the full details are known up front to the developer, in which case they can simply write a complete implementation.
+
+However, sometimes the developer may not know the final data source details.
+
+Integration templates bridge the gap between developers and WordPress instance admins when implementing custom data fetching solutions. These templates support scenarios where developers need to create custom blocks that interact with third-party APIs, but may not have full access to authentication details managed by other teams.
```

Comment:
There are other (probably more likely?) scenarios here, for example:

1. Variable data sources, concrete queries. I commit my queries to code, but I define different data sources in the UI for my production vs staging vs local environments. 
2. Supported data sources, unsupported queries. I like using UI to define data sources, so I do so. However, RDB doesn't yet support generating queries automatically for my data source, so I must commit code.
   - Or, I require code-level customization to my queries that requires committing queries to code.

Reply:
Yeah, I think we want to frame this as "You write your code, then provide your variables through the settings page" and the alternative of creating your own very custom data source is kind of "hidden" in "Advanced Usage" or something only when you really need to use a very specific configuration setup.

I don't think providing connection details in the settings page feels like a "mixed" or "separated" approach... it's just making a few key details into variables.

Reply:
So I guess what I'm saying is it might be better to surface this doc as the "primary" flow and label the "fully code based" one where you provide your own credentials in the code as the "advanced usage" or secondary flow.

Does that make sense?